api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -35,4 +37,9 @@ message UdpProxyConfig {
   // The idle timeout for sessions. Idle is defined as no datagrams between received or sent by
   // the session. The default if not specified is 1 minute.
   google.protobuf.Duration idle_timeout = 3;
+
+  // Optional configuration for UDP proxy hash policy. If hash_policy is not set, the hash-based
+  // load balancing algorithms will select a host randomly. Currently the number of hash policies is
+  // limited to 1.
+  repeated type.v3.HashPolicy hash_policy = 4 [(validate.rules).repeated = {max_items: 1}];
```

Comment:
As I mentioned, please add a wrapper message called something like `UdpDatagramHashPolicy`. Inside of that message, have a oneof for the policy type, with `type.v3.HashPolicy` being the only policy right now. Later we can add byte range policies and other thing specific to UDP and QUIC. cc @ggreenway.

Reply:
That is great idea!! Ok. I will add a new wrapper message for that. :)

Reply:
I have a doubt. Which thing did you mean? Please see following two messages.

1. 
`UdpHashPolicy` is a new message that only used by udp.

```
message UdpHashPolicy {
  // The source IP will be used to compute the hash used by hash-based load balancing
  // algorithms.
  message SourceIp {
  }

  oneof policy_specifier {
    option (validate.required) = true;

    SourceIp source_ip = 1;
  }
}

message UdpProxyConfig {
  option (udpa.annotations.versioning).previous_message_type =
      "envoy.config.filter.udp.udp_proxy.v2alpha.UdpProxyConfig";

  // The stat prefix used when emitting UDP proxy filter stats.
  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];

  oneof route_specifier {
    option (validate.required) = true;

    // The upstream cluster to connect to.
    string cluster = 2 [(validate.rules).string = {min_bytes: 1}];
  }

  // The idle timeout for sessions. Idle is defined as no datagrams between received or sent by
  // the session. The default if not specified is 1 minute.
  google.protobuf.Duration idle_timeout = 3;

  // Optional configuration for UDP proxy hash policy. If hash_policy is not set, the hash-based
  // load balancing algorithms will select a host randomly. Currently the number of hash policies is
  // limited to 1.
  repeated type.v3.UdpHashPolicy hash_policy = 4 [(validate.rules).repeated = {max_items: 1}];
}
```


2. 
`HashPolicy` is the already existing message.
```
message UdpDatagramPolicy {
  // Optional configuration for UDP proxy hash policy. If hash_policy is not set, the hash-based
  // load balancing algorithms will select a host randomly. Currently the number of hash policies is
  // limited to 1.
  repeated type.v3.HashPolicy hash_policy = 1 [(validate.rules).repeated = {max_items: 1}];
}

message UdpProxyConfig {
  option (udpa.annotations.versioning).previous_message_type =
      "envoy.config.filter.udp.udp_proxy.v2alpha.UdpProxyConfig";

  // The stat prefix used when emitting UDP proxy filter stats.
  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];

  oneof route_specifier {
    option (validate.required) = true;

    // The upstream cluster to connect to.
    string cluster = 2 [(validate.rules).string = {min_bytes: 1}];
  }

  // The idle timeout for sessions. Idle is defined as no datagrams between received or sent by
  // the session. The default if not specified is 1 minute.
  google.protobuf.Duration idle_timeout = 3;


  // Optional configuration for UDP proxy policy. It will have many polices but it has only hash policy now.
  type.v3.UdpDatagramPolicy policy = 4;
}
```

Reply:
More like (2), but instead of `repeated type.v3.HashPolicy`, put it in a `oneof`, not-repeated.

Reply:
Yes, (2). Your repeated would go on the policy field (make it policies). Then inside each policy you have a oneof. There is where we can add byte range hashing later.

Reply:
Ok. I got your point. I will let you know once api is ready.

Reply:
I have update the api. I hope it is what you expected. Please let me know if it is different. :)